### PR TITLE
Straight port of :c.i into IEx.Helpers

### PR DESCRIPTION
**Pull request by knewter**
I definitely think we ought to modify this to be our own IEx version...but this commit makes `i` work as a direct port of the erlang :c module's code.  Maybe a good starting point for our own?  Not necessarily suggesting it get merged yet - just a call for suggestions. /cc @rvirding

### CODE REVIEW

**Comment by knewter** [comment]
woops, the 'c to continue, q to quit' bit doesn't work it seems

**Comment by knewter** [comment]
I definitely intend to squash this before asking for a real merge fwiw

File: lib/iex/lib/iex/helpers.ex [hunk 1]
```diff
@@ -114,6 +115,155 @@ defmodule IEx.Helpers do
...
+    i_header()
+    paged_i(processes, {0,0,0,0}, num, 50)
+  end
+
+  defp i_header() do
+    iformat_ansi('Pid', 'Initial Call', 'Heap', 'Reds', 'Msgs', [:bright, :yellow])
+    iformat_ansi('Registered', 'Current Function', 'Stack', '', '', [:bright, :yellow])
+  end
+
+  def paged_i([], {r,m,h,s}, _, _) do
```
Comment (josevalim) [inline lib/iex/lib/iex/helpers.ex]: Those should all be private functions. :)

**Comment by knewter** [comment]
@josevalim done.  See anything else / any other requests?

**Comment by josevalim** [comment]
Not yet. I will look at it with care so I can give more feedback, thanks. But just later. :(

**Comment by josevalim** [comment]
Hey @knewter, some feedback:

1. If you have done a straight port of the Erlang code, we likely need to use the Erlang license. Laws are blurry here, so I would avoid it. It would be better if we come up with our implementation

2. Another reason for rewriting is that there are some issues in the erlang implementation that show up with Elixir names, which are usually larger in names than the erlang ones. It would be nice if the results actually adapted to the screen (instead of fixed length like in Erlang) or, alternatively, that we always showed the name fully, without trimming them.

3. I would also move the implementation to a separate module where we format everything and do the IO work just later on. It would be much easier to test. :)

For those reasons, I am closing this issue. I don't think we can merge it at all because of 1. Thank you!

**Comment by eksperimental** [comment]
@josevalim now that otp 18 is released under the same license as Elixir. May I ask you what is that you are trying to avoid by using Erlang code?

**Comment by josevalim** [comment]
@eksperimental we still support otp 17.

**Pull request closed without merging**